### ISSUE TITLE
Theme Showcase: Return to Correct Tab and Scroll Position After Previewing a Theme

### ISSUE THREAD

**Issue by Copons** [Automattic/wp-calypso#84750]
#### Steps to reproduce

1. Starting at URL: `https://wordpress.com/themes/{SITE}`.
2. Navigate to a filtered tab, e.g. select "Blog" and "Free": `https://wordpress.com/themes/free/filter/blog/{SITE}`.
3. Scroll down a bit, choose a theme (e.g. "Ueno"), and click "Preview & Customize" from its more menu.
4. Once in the theme preview screen, click on the Back arrow.

#### What I expected

To return to the Theme Showcase at the correct filter (`https://wordpress.com/themes/free/filter/blog/{SITE}`), and scrolled to the Ueno theme.

#### What happened instead

I returned to the main Theme Showcase page (`https://wordpress.com/themes/{SITE}`), scrolled to the top.

#### Context / Source

This should be solvable by providing two details to the theme preview:
- A return URL, e.g. `/themes/free/filter/blog/{SITE}`.
- A `bookmarkRef` with the theme slug, to scroll back to the theme. (A simple search for `bookmarkRef` in Calypso should give a good idea of how it works).

**Comment by filipanoscampos** [comment]
📌 REPRODUCTION RESULTS

– Tested on Simple: Replicated 

– Tested on Atomic:  Replicated

📌 VIDEO

Reproduced exactly as described above

https://github.com/Automattic/wp-calypso/assets/2540430/3f956235-ed8a-4efa-9cfd-861da3caab17

it would be great if the users were taken to exactly the theme they were previewing to improve user experience.

📌 ACTIONS

- Report kept.

**Comment by rcrdortiz** [comment]
We updated the theme details page some time ago. Now we don't have a back button. The issue is reproducible if you reload the theme details page, the main problem is that the bookmark ref isn't being persisted in the store. I've created a POC for enabling persistence, but the problem is that once we have a bookmark ref in the store, and we navigate to the TS (or load it), the bookmark applies regardless of whether we navigate back.
